python/src/scipp/plot/plot_1d.py

```diff
@@ -99,6 +98,35 @@ def __init__(self,
                          config.plot.height / config.plot.dpi),
                 dpi=config.plot.dpi)
 
+        # Determine whether error bars should be plotted or not
+        self.variances = {}
+        for name, var in self.scipp_obj_dict.items():
+            self.variances[name] = var.variances is not None
+        if variances is not None:
+            if isinstance(variances, bool):
+                for name, var in self.scipp_obj_dict.items():
+                    self.variances[
+                        name] = variances and var.variances is not None
+            elif isinstance(variances, dict):
+                for name, v in variances.items():
+                    if name in self.scipp_obj_dict:
+                        self.variances[
+                            name] = variances[name] and self.scipp_obj_dict[
+                                name].variances is not None
+                    else:
+                        print("Warning: key {} was not found in list of "
+                              "entries to plot and will be ignored.".format(
+                                  name))
+            elif isinstance(variances, list):
+                keys = list(sorted(self.scipp_obj_dict.keys()))
+                for i, v in enumerate(variances):
+                    self.variances[
+                        keys[i]] = variances[i] and self.scipp_obj_dict[
+                            keys[i]].variances is not None
```

Comment:
So this just accepts a list of `[True, False, ...]`? IS this really so useful, to justify all the extra code?

Reply:
I guess it's not very useful and actually error-prone. I'm supporting just a list of colors and markers, as I've often found this to be useful when plotting many curves and not really needing full control over the colors (as one would do in a waterfall plot), but I agree that for the variances you need to be more explicit.